test/e2e/volumes.go

```diff
@@ -161,17 +162,35 @@ func startVolumeServer(client clientset.Interface, config VolumeTestConfig) *v1.
 			Volumes: volumes,
 		},
 	}
-	serverPod, err := podClient.Create(serverPod)
-	framework.ExpectNoError(err, "Failed to create %s pod: %v", serverPod.Name, err)
 
+	var pod *v1.Pod
+	serverPod, err := podClient.Create(serverPod)
+	// ok if the server pod already exists. TODO: make this controllable by callers
+	if err != nil {
+		if apierrs.IsAlreadyExists(err) {
+			framework.Logf("Ignore \"already-exists\" error, re-get pod...")
+			By(fmt.Sprintf("re-getting the %q server pod", serverPodName))
+			serverPod, err = podClient.Get(serverPodName, metav1.GetOptions{})
+			framework.ExpectNoError(err, "Cannot re-get the server pod %q: %v", serverPodName, err)
+			pod = serverPod
+		} else {
+			framework.ExpectNoError(err, "Failed to create %q pod: %v", serverPodName, err)
+		}
+	}
 	framework.ExpectNoError(framework.WaitForPodRunningInNamespace(client, serverPod))
 
-	By("locating the server pod")
-	pod, err := podClient.Get(serverPod.Name, metav1.GetOptions{})
-	framework.ExpectNoError(err, "Cannot locate the server pod %v: %v", serverPod.Name, err)
+	if pod == nil {
+		By(fmt.Sprintf("locating the %q server pod", serverPodName))
+		pod, err = podClient.Get(serverPodName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "Cannot locate the server pod %q: %v", serverPodName, err)
+		//The server pod doesn't do a mount and with the WaitForPod... above the server pod is
+		//running, therefore the sleep below should not be necessary.
+		//TODO: if a delay is actually required it is better to create a test pod which
+		//      mounts the volume and then wait for that pod to terminate.
+		//By("sleeping a bit to give the server time to start")
```

Comment:
A running pod does not indicate that a daemon is actually running inside it. For NFS, the slowdown is negligible, other test images do non-trivial initialization on startup (e.g. gluster and ceph RBD).  I think that kubelet should retry mounting a volume when the server is not available for the first time, however can you post a separate PR with this so it's easier to revert when something goes wrong?